packages/jaeger-ui/src/components/TracePage/TracePageHeader/TracePageSearchBar.css

```diff
@@ -15,11 +15,19 @@ limitations under the License.
 */
 
 .TracePageSearchBar {
-  max-width: 20rem;
+  width: 40%;
+}
+
+.TracePageSearchBar--InputGroup {
+  justify-content: flex-end;
 }
 
 .TracePageSearchBar--bar {
-  display: flex;
+  max-width: 100%;
+  transition: max-width 1.5s;
+}
+
+.TracePageSearchBar--bar:not(:focus-within) {
```

Comment:
Why prefer the use of `:not(...)` over:

```css
.TracePageSearchBar--bar {
  max-width: 20rem;
  transition: max-width 0.5s;
}

.TracePageSearchBar--bar:focus-within {
  max-width: 100%;
}
```

Also, the `1.5s` seems a bit slow?

Reply:
the `:not(...)` was a relic of an earlier approached, updated :+1:
Changed to 0.5s and seems fine. Initially the width delta was larger and when it was too fast it felt instantaneous rather than smooth.